### PR TITLE
Code Quality: Fixed an issue where folder selection dialog wouldn't work

### LINKED ISSUE #15550: Bug: Crash when extracting a folder to a OneDrive folder (Business or personal)
### Description

The app crash systematically when extracting a folder from download to a folder in a OneDrive folder (personal or Business)

### Steps To Reproduce

1. Open files app
2. extract a full folder that you downloaded in a OneDrive folder
3. See the crash

### Requirements

- Fix crash that occurs when extracting a folder to a folder in OneDrive

### Files Version

3.4.4.0

### Windows Version

26120.751 (24H2)

### Log File

[debug.log](https://github.com/user-attachments/files/15566348/debug.log)

### PR DESCRIPTION
**Pull request by hishitetsu**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes #15550 

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Right-click an archive.
2. Select Extract -> Extract files.
3. Click the Browse button and select a destination folder.
4. It no longer crashes and can be extracted normally.

### CODE REVIEW

File: src/Files.App/Views/Properties/LibraryPage.xaml.cs [hunk 1]
```diff
@@ -107,7 +95,7 @@ private async Task AddLocationAsync()
 			if (!result)
 				return;
 
-			var folder = await StorageHelpers.ToStorageItem<StorageFile>(filePath);
+			var folder = await StorageHelpers.ToStorageItem<BaseStorageFolder>(filePath);
```
Comment (hishitetsu) [inline src/Files.App/Views/Properties/LibraryPage.xaml.cs:98]: This is another place that had the same problem.

Comment (0x5bfa) [reply src/Files.App/Views/Properties/LibraryPage.xaml.cs:98]: Yes. I saw this method with StorageFile was causing cast error in export/import settings command.
Should we basically use BaseStorageFolder until we fully migrate to Files.App.Storage?

**Review by 0x5bfa — APPROVED** [review]
Thank you so much. I didn’t check code after user select files, my bad…
Code wise, looks pretty good．

File: src/Files.App/Views/Properties/LibraryPage.xaml.cs (hunk 1 above)
Comment (hishitetsu) [reply src/Files.App/Views/Properties/LibraryPage.xaml.cs:98]: Yes, otherwise it wouldn't work.

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**